test/constructors.jl

```diff
@@ -1,5 +1,5 @@
 module TestConstructors
-    using Base.Test
+    using Base.Test, Compat
```

Comment:
With the change, you don't need Compat AFAICT. Other than that, looks good to merge. Thanks for doing this!